boltdesignsystem/bolt
# Fix Button `disabled` prop

## Jira

http://vjira2:8080/browse/BDS-1993

## Summary

Fixes button `disabled` prop, adds test coverage.

## Details

When we switched Button over to LitElement we did not add `reflect: true` to the `disabled` prop. This is required for boolean HTML attributes to be reflected as internal props. Otherwise, they will return an empty string or `undefined`. This PR adds the `reflect` option and two tests to prevent future regressions.

As a general rule, when using LitElement, boolean schema props must be configured with `reflect: true`. This allows external changes to the attribute to be reflected in the internal prop and vice versa.

If the prop is internal-only (see [dotted](https://github.com/boltdesignsystem/bolt/blob/master/packages/components/bolt-tooltip/src/tooltip.js#L20), [expanded](https://github.com/boltdesignsystem/bolt/blob/master/packages/components/bolt-band/src/band.js#L7)) there's no need to use `reflect`. The reason we still add these as internal props (even though they're not in the schema) is because we can leverage LitElement's built-in observers to do something when that prop value changes or to automatically trigger a re-render.

## How to test

- Review files changed.
- Check `/pattern-lab/?p=components-button-style-variations` are verify disabled buttons have the proper semi-transparent style.
